### PR TITLE
Added a parse_query_string() example

### PR DESCRIPTION
**Pull request by vitor-alves**
Hi

I added one more example that uses  `parse_query_string()` to parse the GET request parameters.
I think it might be useful for newcomers. It's a pretty common thing to do and I needed it in one of my projects and couldn't find any examples.

The following GET to path `/parse` with parameters `name=Mary` and `lastName=Smith`

`localhost:8080/parse?name=Mary&lastName=Smith`

will be parsed with `parse_query_string()` and the `name + lastName` will be returned as response.

Its similar to the _John Smith_  example, but instead of parsing a JSON the `query_string` is parsed.

I also fixed a few typos here and there. 

Please consider this pull request, and thanks for this project. I have been using it a lot.

### CODE REVIEW

**Comment by eidheim** [comment]
Thank you, I'll look through the changes Today hopefully.

File: http_examples.cpp [hunk 1]
```diff
@@ -67,7 +67,7 @@ int main() {
                 << name;
     }
     catch(const exception &e) {
-      *response << "HTTP/1.1 400 Bad Request\r\nContent-Length: " << strlen(e.what()) << "\r\n\r\n"
+      *response << "http/1.1 400 bad request\r\ncontent-length: " << strlen(e.what()) << "\r\n\r\n"
```
Comment (eidheim) [inline http_examples.cpp]: The capital letters should be kept here?

File: http_examples.cpp [hunk 2]
```diff
@@ -142,7 +164,7 @@ int main() {
       // Check if path is within web_root_path
       if(distance(web_root_path.begin(), web_root_path.end()) > distance(path.begin(), path.end()) ||
          !equal(web_root_path.begin(), web_root_path.end(), path.begin()))
-        throw invalid_argument("path must be within root path");
+        throw invalid_argument("Path must be within root path");
```
Comment (eidheim) [inline http_examples.cpp]: Exception messages typically starts with lower case

File: http_examples.cpp [hunk 3]
```diff
@@ -131,6 +131,28 @@ int main() {
...
+      *response << "HTTP/1.1 200 OK\r\n"
+                << "Content-Length: " << name.length() << "\r\n\r\n"
+                << name;
+    }
+    catch(const exception &e) {
+      *response << "http/1.1 400 bad request\r\ncontent-length: " << strlen(e.what()) << "\r\n\r\n"
+                << e.what();
+    }
+  };
+
```
Comment (eidheim) [inline http_examples.cpp]: One idea could be to instead add the query string parsing to the `/info` request example.

File: http_examples.cpp (hunk 1 above)
Comment (vitor-alves) [reply http_examples.cpp]: Yeah, they should be kept.
My mistake

File: http_examples.cpp (hunk 2 above)
Comment (vitor-alves) [reply http_examples.cpp]: Hmm
I made them upper case to match the std::cerr messages that start with upper case letters, but did not know they should be lower case. Revert this please.
Thanks for the tip

File: http_examples.cpp (hunk 3 above)
Comment (vitor-alves) [reply http_examples.cpp]: Yeah, sure. Feel free to modify it however you want. Adding it to /info seems like a good idea. 
The important thing is to show in any of the examples the use of parse_query_string() so people will know it exists and how to use it.

**Comment by eidheim** [comment]
Thank you again. I pushed this with some changes in https://github.com/eidheim/Simple-Web-Server/commit/b9d4be229f41e3eda72394451d479f5245d2a600.

**Pull request closed without merging**